plugin.video.psvue/resources/lib/globals.py

```diff
@@ -0,0 +1,153 @@
+import threading
+import time, math
+import requests
+import random
+import urllib
+import xbmc, xbmcgui, xbmcaddon, xbmcvfs, xbmcplugin
+import _strptime
+
+
+if not xbmc.getCondVisibility('System.HasAddon(pvr.iptvsimple)'):
+    dialog = xbmcgui.Dialog()
+    dialog.notification('PS Vue EPG', 'Please enable PVR IPTV Simple Client', xbmcgui.NOTIFICATION_INFO, 5000, False)
+    sys.exit()
+
+ADDON = xbmcaddon.Addon()
+PS_VUE_ADDON = xbmcaddon.Addon('plugin.video.psvue')
+ADDON_PATH_PROFILE = xbmc.translatePath(PS_VUE_ADDON.getAddonInfo('profile'))
+UA_ANDROID_TV = 'Mozilla/5.0 (Linux; Android 6.0.1; Hub Build/MHC19J; wv) AppleWebKit/537.36 (KHTML, like Gecko)' \
+                ' Version/4.0 Chrome/61.0.3163.98 Safari/537.36'
+UA_ADOBE = 'Adobe Primetime/1.4 Dalvik/2.1.0 (Linux; U; Android 6.0.1 Build/MOB31H)'
+CHANNEL_URL = 'https://media-framework.totsuko.tv/media-framework/media/v2.1/stream/channel'
+EPG_URL = 'https://epg-service.totsuko.tv/epg_service_sony/service/v2'
+SHOW_URL = 'https://media-framework.totsuko.tv/media-framework/media/v2.1/stream/airing/'
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+IPTV_SIMPLE_ADDON = xbmcaddon.Addon('pvr.iptvsimple')
+SAVE_LOCATION = ADDON_PATH_PROFILE
+COPY_LOCATION = xbmc.translatePath(ADDON.getSetting(id='location'))
+VERIFY = True
+
+
+def get_json(url):
+    headers = {
+        'Accept': '*/*',
+        'reqPayload': PS_VUE_ADDON.getSetting(id='EPGreqPayload'),
+        'User-Agent': UA_ANDROID_TV,
+        'Accept-Encoding': 'gzip, deflate, br',
+        'Accept-Language': 'en-US,en;q=0.5',
+        'X-Requested-With': 'com.snei.vue.android',
+        'Connection': 'keep-alive'
+    }
+
+    r = requests.get(url, headers=headers, cookies=load_cookies(), verify=VERIFY)
+
+    if r.status_code != 200:
+        dialog = xbmcgui.Dialog()
+        msg = 'The request could not be completed.'
+        try:
+            json_source = r.json()
+            msg = json_source['header']['error']['message']
+        except:
+            pass
+        dialog.notification('Error ' + str(r.status_code), msg, xbmcgui.NOTIFICATION_INFO, 9000)
+        sys.exit()
+    return r.json()
+
+
+def load_cookies():
+    cookie_file = os.path.join(ADDON_PATH_PROFILE, 'cookies.lwp')
+    cj = cookielib.LWPCookieJar()
+    try:
+        cj.load(cookie_file, ignore_discard=True)
+    except:
+        pass
+    return cj
+
+
+def find(source, start_str, end_str):
+    start = source.find(start_str)
+    end = source.find(end_str, start + len(start_str))
+    if start != -1:
+        return source[start + len(start_str):end]
+    else:
+        return ''
+
+
+def string_to_date(string, date_format):
+    try:
+        date = datetime.strptime(str(string), date_format)
+    except TypeError:
+        date = datetime(*(time.strptime(str(string), date_format)[0:6]))
+    return date
+
+
+def check_iptv_setting(id, value):
+    if IPTV_SIMPLE_ADDON.getSetting(id) != value:
+        IPTV_SIMPLE_ADDON.setSetting(id=id, value=value)
```

Comment:
Can't set another plugin's settings, you could set up a button to opensettings().

Reply:
Normally I would strongly agree with this rule but in this case I don't think it should apply. First this is a "opt-in" option and I'd argue that I am using IPTV Simple Client the way it is intended, by supplying it with a playlist & xmltv path. Nothing malicious or overstepping, I am simply automating a step that I believe most users wouldn't be able to or care to do on their own. The idea of popping a message and telling them how to do it instead of just doing it and knowing it is done correctly seems silly to me. As the former introduces user error that will inevitably fall back on the dev having to lend support to.

Reply:
I don't believe the repository has ever made an exception when it comes to writing to the settings of another plugin. IMO the whole IPTV simple thing is a bit hackish. Generating M3Us and XMLTV files are fine, manipulating Kodis api to shoehorn support seems problematic.

Reply:
![image](https://user-images.githubusercontent.com/8268725/66403494-aa2d6e80-e9b4-11e9-8cd2-3bad7fba5722.png)
@enen92  @Rechi 
I'd like to request access to write to IPTV Simple settings in order to allow users to use PS Vue through the built-in Kodi EPG. This is a "opt-in" option.

Reply:
@eracknaphobia I don't think this situation applies; that statement applies to addon folders and writing files to them. ie Cache modules, etc. However,  it doesn't hurt to revisit the issue.

Reply:
@eracknaphobia would you consider a yes/no before changing the setting? 
I'm fine with it as it is though.

Reply:
@enen92 Yes, I have no problem doing that.

Reply:
I'd also prefer that the code is not so generic and that it identifies clearly what settings you're trying to manipulate. I douby you need such a generic method.
what exactly do you need to do?

Reply:
@enen92 
Basically I'm setting the option for playlist and epg location to local and setting the paths to the files in the ps vue addon_data folder. I'm also setting the logos to be pulled remotely and to prefer the logo urls in the m3u file.

This section should make it clearer as to what values are set.
https://github.com/xbmc/repo-plugins/blob/b9d2b3b1d36aebaa9b1288d590cc2b191d48c019/plugin.video.psvue/resources/lib/globals.py#L148-L153
https://github.com/xbmc/repo-plugins/blob/b9d2b3b1d36aebaa9b1288d590cc2b191d48c019/plugin.video.psvue/resources/lib/database.py#L140